omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleResource1_8.java

```diff
@@ -192,4 +199,43 @@ public Object upload(MultipartFile file, RequestContext context) throws Response
 			}
 		}
 	}
+	
+	private void startModules(Collection<Module> modules, Module existingModule, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		if (modules.size() > 1) {
+			modules = moduleFactoryWrapper.getModulesInStartupOrder(modules);
+		}
+		
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStopped(module) && module.getModuleId() != existingModule.getModuleId()) {
+				needsRefresh = moduleFactoryWrapper.startModuleSkipRefresh(module, servletContext) || needsRefresh;
+			}
+		}
+		//check if any module has been started, doesn't refresh WAC if all modules failed to start
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+		
+		findAndThrowStartupErrors(modules);
+	}
+	
+	private void findAndThrowStartupErrors(Collection<Module> modules) {
+		List<Exception> errors = new ArrayList<Exception>();
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				//module actions are executed in other thread, so we need to explicitly check and throw them
+				if (module.getStartupErrorMessage() != null) {
+					errors.add(new ModuleException(module.getStartupErrorMessage()));
+				}
+			}
+		}
+		
+		if (!errors.isEmpty()) {
+			StringBuilder stringBuilder = new StringBuilder();
+			for (Exception error : errors) {
+				stringBuilder.append(error.getMessage()).append("; ");
+			}
+			throw new RuntimeException(stringBuilder.toString());
```

Comment:
Why throw a runtime exception instead of module exception?

Reply:
Have changed it to use module exception